src/linker/Linker/Annotations.cs

```diff
@@ -678,6 +678,21 @@ public void EnqueueVirtualMethod (MethodDefinition method)
 			if (!method.IsVirtual)
 				return;
 
+			// Implementations of static interface methods are not virtual and won't reach here
```

Comment:
I actually removed the implementations of static interface methods from _virtual_methods, so it should only be methods that are actually 'virtual' in IL in the list. Do you think it's worth it to add both the base and the implementations?

Reply:
Adding them in MarkBaseMethods is necessary for static methods in a preserved scope. Otherwise we miss annotation checks and marking the implementations as "Base is in a preserved scope".

Reply:
I've added some tests [here](https://github.com/dotnet/linker/pull/2926/files#diff-78d519640583fdd90cf7cf7569c8da2f43baf94d2a8b9aa91460a1d43a5f78aaR738) that fail when the line is removed. I also found that instance interface methods aren't checked for matching annotations if the base in a preserved scope, so I removed the IsStatic check. This may warn if the interface implementation is removed or if the static interface methodImpl is removed, but I think it makes sense to still warn there.